File: Baekjoon/C/Edu/18258.c
#2022.7.6 18258 큐 c
#include <stdio.h>
#include <string.h>
int main(void){
    int n;
    int num;
    int front=0, rear=-1;
    int stack[2000001];
    char say[10];
    scanf("%d",&n);
    for(int i=0;i<n;i++){
        
        scanf("%s",say);
        if(strcmp(say,"push")==0){
            scanf("%d", &num);
            stack[++rear]=num;
        
        }
        else if(strcmp(say,"pop")==0){
            if(rear-front+1==0){
                printf("-1\n");
            }
            else{
                printf("%d\n", stack[front++]);
            }
            
        }
        else if(strcmp(say,"size")==0){
            printf("%d\n",rear-front+1);
        }
        else if(strcmp(say,"empty")==0){
            if(rear-front+1==0){
                printf("1\n");
            }
            else{
                printf("0\n");
            }
        }
        else if(strcmp(say,"front")==0){
            if(rear-front+1==0){
                printf("-1\n");
            }
            else{
                printf("%d\n",stack[front]);
            }
            
        }
        else if(strcmp(say,"back")==0){
            if(rear-front+1==0){
                printf("-1\n");
            }
            else{
                printf("%d\n",stack[rear]);
            }
            
        }
        for(int i=0;i<10;i++){
            say[i]='\0';
        }
        
    }
    return 0;
}

File: Baekjoon/Python/Edu/18258.py
#2022.7.6 18258 큐 python
import sys
from collections import deque
input=sys.stdin.readline

n=int(input())
test=deque([])
def queue(how):
    if how[0]=='push':
        test.append(how[1])
    elif how[0]=='pop':
        if len(test)==0:
            print(-1)
        else:
            print(test.popleft())
    elif how[0]=='size':
        print(len(test))
    elif how[0]=='empty':
        if len(test)==0:
            print(1)
        else:
            print(0)
    elif how[0]=='front':
        if len(test)==0:
            print(-1)
        else:
            print(test[0])
    elif how[0]=='back':
        if len(test)==0:
            print(-1)
        else:
            print(test[len(test)-1])
for i in range(n):
    how=input().split()
    queue(how)
